### ISSUE TITLE
aws_apigateway: LambdaRestApi does not respect the proxy: false property

### ISSUE THREAD

**Issue by jglesner** [aws/aws-cdk#21787]
### Describe the bug

Hello!  I want to use the LambdaRestApi construct and use a non-proxy integration.  Therefore, I set the proxy prop to false as shown in the [LambdaRestApi documentation](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigateway.LambdaRestApi.html) to create a 

```typescript
   const gateway = new apigateway.LambdaRestApi(
      this,
      "APIGW_MobileApiTestEndpoint",
      {
        handler: mobileApi_test_Lambda,
        endpointConfiguration: {
          types: [apigateway.EndpointType.REGIONAL],
        },
        proxy: false,
      }
    );

   const testResource = gateway.root.addResource("test");
   testResource.addMethod("POST"); 
```

Trouble is, CDK deploys the stack with the API Gateway Integration Request boolean 'Use Lambda Proxy Integration' set to True.  Testing this produces the result you would expect for a Lambda that isn't expecting a proxy response ... it fails.  When testing that Resource using the console, the API Gateway Test Log also shows that it's sending the full proxy payload. You can plainly see this in the console: 

<img width="412" alt="Screen Shot 2022-08-26 at 7 39 50 PM" src="https://user-images.githubusercontent.com/1482774/187004359-f3643935-3d5f-404e-946e-874f36f58d46.png">

This bug was reported back in February 2021 in issue [#13026](https://github.com/aws/aws-cdk/issues/13026).  At that time @nija-at tested it, and said it was working normally.  However, in his test he actually demonstrated that this is a bug ... he just didn't seem to realize it.  In his sample nodejs code, @nija-at is returning the status code in the response `return { statusCode: 200, body: "Hello world!" }` which is what you have to do when the lambda integration IS configured to be a proxy, even though he set Proxy to false in his test.  In a non-proxy lambda integration, you don't return the status code.  Others have reported this bug on ticket 13026 as well.  

I would greatly appreciate it if we can get a fix for this! 

### Expected Behavior

I expect the 'proxy: false' property to be respected by the LambdaRestApi construct.

### Current Behavior

The 'proxy: false' property of the LambdaRestApi construct is not respected. 

### Reproduction Steps

Shown in the description above.

### CDK CLI Version

2.39.0

### Framework Version

2.39.0

### Node.js Version

14.17.0

### OS

OS X Monterey 12.1

### Language

Typescript

### Language Version

3.9.7

**Comment by peterwoodworth** [comment]
Synthing the code provided, it generates a Method resource with [Integration type AWS_PROXY](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apitgateway-method-integration.html#cfn-apigateway-method-integration-type)

The code for LambdaRestApi generates a LambdaIntegration, and doesn't change what gets passed in for `integrationOptions.Proxy`, which defaults to true no matter what you set for the proxy prop on LambdaRestApi.

https://github.com/aws/aws-cdk/blob/e36bfe5fe17d8e4e4a31903c95c9482346cf99ba/packages/%40aws-cdk/aws-apigateway/lib/lambda-api.ts#L51-L58

So in short, i think we will need to ensure the LambdaIntegration created is created with proxy: false as well. You can set this with options currently, all that setting integrationOptions.Proxy to false will do is change `AWS_PROXY` in the template to `AWS`. Does this give the desired behavior?

**Comment by jglesner** [comment]
Thank you for the quick reply @peterwoodworth.  I just tested that out -- the lambda integration was converted to a non-proxy lambda.  However, there are no Method Responses or Integration Responses set up, so even a success response from the lambda is converted to a 500 response from API Gateway.   

<img width="1166" alt="Screen Shot 2022-08-26 at 10 52 07 PM" src="https://user-images.githubusercontent.com/1482774/187012161-d059aa2c-aa60-4833-98ef-513fd561c59b.png">

Not sure if this is a by-product of converting an API GW Integration from proxy to non-proxy.  The [documentation](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigateway.LambdaRestApi.html) doesn't mention having to define Method Responses or Integration Responses; this might need to be clarified as necessary.   

Looking at the [integrationOptions documentation](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigateway.LambdaRestApiProps.html), it appears possible to set up `defaultMethodOptions` and `integrationResponses` within the `integrationOptions` block.  Will give this a try.

**Comment by peterwoodworth** [comment]
Let me know if you can get something working with `integrationOptions` @jglesner!

**Comment by jglesner** [comment]
@peterwoodworth, I spent some serious time today trying to get the IntegrationRequest, IntegrationResponse, and MethodResponse all configured correctly to work for a non-proxy method.  I finally found one way to make it all work as intended.  But this is pretty convoluted, and I think there is another bug here.   

First, what didn't work.  As I said in my post above, I was hoping to use the [LambdaRestApiProps](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigateway.LambdaRestApiProps.html), it appears possible to set up `defaultMethodOptions` and `integrationOptions` to configure the IntegrationRequest, IntegrationResponse, and MethodResponse. While `integrationOptions` works as intended, unfortunately, `defaultMethodOptions` seems to be partially ignored.  Despite defining the `methodResponses`, it was not added to the MethodResponse.

```typescript
  const gateway = new apigateway.LambdaRestApi(
      this,
      "APIGW_MobileApiTestEndpoint",
      {
        handler: mobileApi_test_Lambda,
        endpointConfiguration: {
          types: [apigateway.EndpointType.REGIONAL],
        },
        proxy: false, // probably not needed until this gets fixed
        integrationOptions: {
           proxy: false,
           integrationResponses: [{ statusCode: "200" }],
           passthroughBehavior: PassthroughBehavior.WHEN_NO_TEMPLATES,
      }
      defaultMethodOptions: {
         methodResponses: [
         {
             statusCode: "200",
             responseModels: {
               "application/json": apigateway.Model.EMPTY_MODEL,
             },
          },
        ],
      },
    }
  );
```

To get this to work as intended, I needed to move my attempts to configure the IntegrationRequest, IntegrationResponse, and MethodResponse out of the LambdaRestApi altogether.  As you'll notice below, I had to configure a `LambdaIntegration`, and configure the `methodResponses` when adding the `POST` method to the resource.  Here is the code that finally worked. 

```typescript
    const gateway = new apigateway.LambdaRestApi(
      this,
      "APIGW_MobileApiTestEndpoint",
      {
        handler: mobileApi_address_Lambda,
        endpointConfiguration: {
          types: [apigateway.EndpointType.REGIONAL],
        },
        proxy: false, // probably not necessary until fixed
      }
   );

   const integration = new apigateway.LambdaIntegration(
      mobileApi_test_Lambda,
      {
        integrationResponses: [{ statusCode: "200" }],
        passthroughBehavior: PassthroughBehavior.WHEN_NO_TEMPLATES,
        proxy: false,
      }
    );

    const addressResource = gateway.root.addResource("test");
    addressResource.addMethod("POST", integration, {
      methodResponses: [
        {
          statusCode: "200",
          responseModels: {
            "application/json": apigateway.Model.EMPTY_MODEL,
          },
        },
      ],
    });
```

**Comment by peterwoodworth** [comment]
Thanks for the thorough investigation here, it's really helpful 🙂 It might take us a while to get to this though. If you have any ideas on how to improve this documentation and submit that in a PR, or even any redesigns, it would be super helpful and much appreciated! Check out our [contributing guide](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md) if you're interested

**Comment by miguelmartin-eb** [comment]
>     const testResource = gateway.root.addResource("test");
>     testResource.addMethod("POST", integration, {
>     methodResponses: [
>      {
>          statusCode: "200",
>         responseModels: {
>           "application/json": apigateway.Model.EMPTY_MODEL,
>        },
>       },
>     ],
>     });

HI there @jglesner ! Did you add any other config for this to work?

I'm currently using the exact same and, despite the method response showing correctly in the console, the api gateway still throws this error :/ 

```
Execution failed due to configuration error: No match for output mapping and no default output mapping configured. Endpoint Response Status Code: 202
```

The only difference I can spot is that my lambda is returning a 202 instead of a 200.

**Comment by ChristianUlbrich** [comment]
@miguelmartin-eb For me this is working **now**, many thanks to @jglesner code example. While the bug persists still today - `proxy: false` basically gets ignored, the problem seems to be more apparent, if you are using a _custom Lambda runtime_, such as `provided.al2023` or `Runtime.PROVIDED_AL2023` in CDK lingo.

If using a "normal" NodeJs runtime, the following:
```typescript
// [...] in a construct...
 const myFunction = new lambda.Function(this, 'HelloWorldFunction', {
      runtime: lambda.Runtime.NODEJS_20_X, // Provide any supported Node.js runtime
      handler: 'index.handler',
      code: lambda.Code.fromInline(`
        exports.handler = async function(event) {
          return {
            statusCode: 200,
            body: JSON.stringify('Hello CDK!'),
          };
        };
      `),
    });

const functionApi = new apigateway.LambdaRestApi(this, 'SimpleHelloWorldApi', {
      handler: myFunction,
      proxy: false,
    });

    const helloResource = functionApi.root.addResource('hello');
    helloResource.addMethod('GET');
```
_works_. It seems, that for _custom runtimes_, you need additional configuration, that gets set by default if using the AWS console for API Gateway, whereas the _default_ from the `LambdaRestApi` does not have the same defaults and on the other hand, internally non-custom runtimes seem to instructed differently in AWS, than custom ones, i.e. they do not need the additional stuff (`integrationResponse`, `methodResponse`) that custom ones need.

When setting up RestAPIs for non-custom runtimes in AWS, they get the same defaults and yet they work, so I think the core problem is, that `LambdaRestApi` does not set the same _defaults_ as one might expect. I think a potential fix might be a **new** _type_, such as `DefaultLambdaRestApi`, that behaves exactly as the AWS console does and would thus allow to work with both custom and non-custom runtimes or at least match expectations better.
